core/src/main/java/io/micronaut/core/type/Argument.java

```diff
@@ -174,6 +174,37 @@ default boolean isAssignableFrom(@NonNull Class<?> candidateType) {
         return getType().isAssignableFrom(Objects.requireNonNull(candidateType, "Candidate type cannot be null"));
     }
 
+    /**
+     * Checks if the argument can be assigned to this argument.
+     * @param candidateArgument The candidate argument
+     * @return True if it is assignable from.
+     * @since 3.0.0
+     */
+    default boolean isAssignableFrom(@NonNull Argument candidateArgument) {
```

Comment:
Should be `Argument<?>` with the wild card

Reply:
Added